algoliasearch-common/src/main/java/com/algolia/search/SyncObjects.java

```diff
@@ -206,6 +205,78 @@ default TaskSingleIndex saveObjects(
     return getApiClient().saveObjects(getName(), objects, requestOptions);
   }
 
+  /**
+   * Replace all objects
+   *
+   * @param objects the list objects to update
+   * @param safe run the method in a safe way
+   */
+  default void replaceAllObjects(@Nonnull Iterable<T> objects, boolean safe)
+      throws AlgoliaException {
+    replaceAllObjects(objects, safe, new RequestOptions());
+  }
+
+  /**
+   * Replace all objects
+   *
+   * @param objects the list objects to update
+   * @param safe run the method in a safe way
+   * @param requestOptions Options to pass to this request
+   */
+  default void replaceAllObjects(
+      @Nonnull Iterable<T> objects, boolean safe, @Nonnull RequestOptions requestOptions)
+      throws AlgoliaException {
+
+    String tmpName = this.getName() + "_tmp_" + UUID.randomUUID().toString();
+    List<String> scope = new ArrayList<>(Arrays.asList("rules", "settings", "synonyms"));
+    List<Task> batchResponses = new ArrayList<>();
+
+    // Copy all index resources from production index
+    Task copyIndexResponse = getApiClient().copyIndex(getName(), tmpName, scope, requestOptions);
+
+    if (safe) {
+      getApiClient().waitTask(copyIndexResponse);
+    }
+
+    // Send records (batched automatically)
+    ArrayList<T> records = new ArrayList<>();
+
+    for (T object : objects) {
+
+      if (records.size() == 1000) {
```

Comment:
Performance issue ( at least on PHP ): You may want to increase a counter instead of calculating the size of the records every object.

Reply:
I've double checked: `ArrayList.size()` is constant-time in Java. Nothing to worry about here ;)

https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html

Reply:
Yes :) Thanks for checking.
Sorry I think I answered this directly on slack.